### PR TITLE
Propagate TypeVarType column in TypeAnalyser

### PR DESCRIPTION
**Pull request by hauntsaninja**
Also use keyword arguments + copy_modified for this in many places

### CODE REVIEW

File: mypy/typeanal.py [hunk 1]
```diff
@@ -1553,13 +1553,14 @@ def anal_type(self, t: Type, nested: bool = True, *, allow_param_spec: bool = Fa
...
-                var_def.variance,
-                var_def.line,
+                name=var_def.name,
+                fullname=var_def.fullname,
+                id=var_def.id.raw_id,
+                values=self.anal_array(var_def.values),
+                upper_bound=var_def.upper_bound.accept(self),
+                variance=var_def.variance,
+                line=var_def.line,
+                column=var_def.column,
```
Comment (hauntsaninja) [inline mypy/typeanal.py:1563]: This is the actual change

File: mypy/checkexpr.py [hunk 2]
```diff
@@ -4742,8 +4742,8 @@ def visit_dictionary_comprehension(self, e: DictionaryComprehension) -> Type:
 
             # Infer the type of the list comprehension by using a synthetic generic
             # callable type.
-            ktdef = TypeVarType("KT", "KT", -1, [], self.object_type())
-            vtdef = TypeVarType("VT", "VT", -2, [], self.object_type())
+            ktdef = TypeVarType("KT", "KT", id=-1, values=[], upper_bound=self.object_type())
```
Comment (ikonst) [inline mypy/checkexpr.py:4745]: Do we want to go all-kw everywhere?

Comment (JelleZijlstra) [reply mypy/checkexpr.py:4745]: I feel like it's a lot clearer.

File: mypy/typeanal.py [hunk 3]
```diff
@@ -344,12 +344,12 @@ def visit_unbound_type_nonoptional(self, t: UnboundType, defining_literal: bool)
                     )
                 # Change the line number
                 return TypeVarType(
```
Comment (JelleZijlstra) [inline mypy/typeanal.py:346]: Could this use copy_modified()?

**Comment by ikonst** [comment]
Can we then solidify it by changing `TypeVarType.__init__` to be kw-only from `values` onwards?

**Comment by hauntsaninja** [comment]
Yeah, I wanted to make `TypeVarType.__init__` use kw-only, but I checked and it would break several plugins. I know we don't promise any compatibility, but it still felt gratuitous.

**Review by JukkaL — APPROVED** [review]
Thanks!

**Pull request merged**